src/ClusterBootstrap/deploy.py

```diff
@@ -2908,6 +2934,23 @@ def run_docker_image( imagename, native = False, sudo = False ):
         else:
             run_docker( matches[0], prompt = imagename, dockerConfig = dockerConfig, sudo = sudo )
 
+def gen_dns_config_script():
```

Comment:
This is a good idea. But please leverage jinja2 templates to create the dns.sh. Also, consider the case for on-perm as well.

Reply:
do you mean generate these scripts in step gen_configs, using utils.render_template/utils.render_template_directory?

Reply:
Yes, please.